### DISCUSSION TITLE
Compilar con Make

### DISCUSSION THREAD

**Discussion by nicoschwend** [IIC2133-PUC/2024-1#61] [Tarea 0] [answered]
Hola, estoy teniendo problemas con make y siento que es algo muy simple jajajaja. Me vi la cápsula varias veces, no modularizé esta tarea, y para testear he estado compilando y corriendo con los siguientes comandos:
valgrind gcc src/ship/main.c -o main
valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./main testx/input.txt output.txt

Como puedo hacer para que funcione con make? es lo único que me falta de la tarea

EDIT: el archivo lo tenia como main y no como ship.c , y no habia hecho make all, pero lo descubrí

**Comment by benjavicente** [comment, marked as answer]
El código inicial funciona con `make`. No había que cambiar el nombre del archivo de entrada `main.c`.

Tiene que funcionar simplemente corriendo `make` en la raíz del repositorio. Lo que te recomiendo ver en lo poco que queda es dejar todo en el archivo `ship/main.c`y ver los errores que te muestra `make`, moviendo funciones arriba si es que te salen que no están definidas pero si existen en el archivo.
